### ISSUE TITLE
Random backup freeze

### ISSUE THREAD

**Issue by QBANIN** [borgbase/vorta#1356]
**Describe the bug**

After Vorta upgrade from 0.8.4 to 0.8.6 backup freeze randomly. 

It looks like it freezes right before backup should finish (after >400GB has been copied over ssh to my backup server).

**To Reproduce**
1. Set up ssh remote repo (compression and/or encryption doesn't matter)
2. Select folder to backup
3. Start backup
4. Wait for files/bytes nr stop to increase but the backup is still ongoing, the only active button is "cancel".

**Environment (please complete the following information):**
- OS: Debian bookworm/sid
- Vorta version: 0.8.6
- Installed from: Debian APT repo

**Additional context**

> 2022-06-12 17:43:52,973 - vorta.keyring.abc - DEBUG - Only available on macOS
> 2022-06-12 17:43:53,009 - vorta.keyring.abc - DEBUG - Using VortaKWallet5Keyring
> 2022-06-12 17:43:53,009 - vorta.borg.borg_job - DEBUG - Using VortaKWallet5Keyring keyring to store passwords.
> 2022-06-12 17:43:53,010 - vorta.keyring.kwallet - DEBUG - Retrieved password for repo ***PRIVACY EDIT****
> 2022-06-12 17:43:53,016 - vorta.borg.jobs_manager - DEBUG - Add job for site 5
> 2022-06-12 17:43:53,016 - vorta.borg.jobs_manager - DEBUG - Start job on site: 5
> 2022-06-12 17:43:53,024 - vorta.borg.borg_job - INFO - Running command /usr/bin/borg create --list --progress --info --log-json --json --filter=AM -C zstd,8 ***PRIVACY EDIT***
> 2022-06-12 17:44:21,275 - vorta.borg.borg_job - INFO - Creating archive at ***PRIVACY EDIT***
> 2022-06-12 17:44:27,630 - vorta.borg.borg_job - INFO - Synchronizing chunks cache...
> 2022-06-12 17:44:27,641 - vorta.borg.borg_job - INFO - Archives: 15, w/ cached Idx: 0, w/ outdated Idx: 0, w/o cached Idx: 15.
> 2022-06-12 17:44:27,654 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-03-30-112615 ...
> 2022-06-12 17:44:32,175 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:32,220 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-04-15-162011 ...
> 2022-06-12 17:44:34,333 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:34,372 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-04-21-234539 ...
> 2022-06-12 17:44:35,823 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:35,863 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-04-28-153008 ...
> 2022-06-12 17:44:37,571 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:37,610 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-01-160654 ...
> 2022-06-12 17:44:38,982 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:39,033 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-05-184950 ...
> 2022-06-12 17:44:40,905 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:40,967 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-13-154945 ...
> 2022-06-12 17:44:42,688 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:42,731 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-15-170610 ...
> 2022-06-12 17:44:44,089 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:44,129 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-16-195618 ...
> 2022-06-12 17:44:45,748 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:45,786 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-17-192529 ...
> 2022-06-12 17:44:47,324 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:47,368 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-19-200825 ...
> 2022-06-12 17:44:48,877 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:48,919 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-23-135113 ...
> 2022-06-12 17:44:50,414 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:50,465 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-29-220751 ...
> 2022-06-12 17:44:52,284 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:52,344 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-05-31-221223 ...
> 2022-06-12 17:44:54,294 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:54,337 - vorta.borg.borg_job - INFO - Fetching and building archive index for DELL-2022-06-01-234614 ...
> 2022-06-12 17:44:56,321 - vorta.borg.borg_job - INFO - Merging into master chunks index ...
> 2022-06-12 17:44:56,392 - vorta.borg.borg_job - INFO - Done.
> 2022-06-12 17:51:22,690 - vorta.scheduler - DEBUG - Refreshing all scheduler timers
> 2022-06-12 17:51:22,698 - vorta.scheduler - DEBUG - Scheduler for profile 1 is disabled.
> 2022-06-12 17:51:22,702 - vorta.scheduler - DEBUG - Scheduler for profile 2 is disabled.

**Comment by elesouef** [comment]
Same here. Debian unstable, Vorta 0.8.6 / Borg 1.2.1. And for me, not as random as the issue occurs on every backup.

**Comment by QBANIN** [comment]
> Same here. Debian unstable, Vorta 0.8.6 / Borg 1.2.1. And for me, not as random as the issue occurs on every backup.

Yeah, it occurs every time but on a random file in my case.

**Comment by elesouef** [comment]
Yes, true. My bad.

**Comment by jplenge** [comment]
Same here on Mac OS  12.4, Vorta 0.8.6 / Borg 1.2.1, installed with homebrew. Every backup freezes near the end of a backup.

**Comment by m3nu** [comment]
Is this another duplicate of https://github.com/borgbase/vorta/issues/1352 ?

**Comment by m3nu** [comment]
If it's the same, please try this test build and I'll do a release after: https://files.qmax.us/vorta/Vorta-0.8.7rc.dmg.zip

Sticking with Borg 1.2.0 for now is the short-term workaround.

**Comment by jplenge** [comment]
The test build solved the problem for me. Thanks!

**Comment by elesouef** [comment]
Cannot test this as I'm not on Mac platform. Ok to test a debian package, though ;-)

**Comment by davidhealey** [comment]
I'm getting what I assume is the same issue on Debian. I notice in Vorta that it hangs and always shows (M) at the end of the file name that it's hanging on.
![image](https://user-images.githubusercontent.com/14021202/173341346-881013f6-9ca8-4e55-9fd4-bd6608771ad1.png)

**Comment by m3nu** [comment]
You can also test the current `master` branch using Pip, as described here: https://vorta.borgbase.com/contributing/#testing-the-current-development-version

**Comment by QBANIN** [comment]
> You can also test the current `master` branch using Pip, as described here: https://vorta.borgbase.com/contributing/#testing-the-current-development-version

> Collecting vorta
  Cloning https://github.com/borgbase/vorta to /tmp/pip-install-mdGxYw/vorta
  Running command git clone -q https://github.com/borgbase/vorta /tmp/pip-install-mdGxYw/vorta
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-mdGxYw/vorta/setup.py'"'"'; __file__='"'"'/tmp/pip-install-mdGxYw/vorta/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-cKAXrc
         cwd: /tmp/pip-install-mdGxYw/vorta/
    Complete output (25 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-mdGxYw/vorta/setup.py", line 3, in <module>
        setup()
      File "/home/qba/.local/lib/python2.7/site-packages/setuptools/__init__.py", line 129, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python2.7/distutils/core.py", line 124, in setup
        dist.parse_config_files()
      File "/home/qba/.local/lib/python2.7/site-packages/setuptools/dist.py", line 494, in parse_config_files
        ignore_option_errors=ignore_option_errors)
      File "/home/qba/.local/lib/python2.7/site-packages/setuptools/config.py", line 106, in parse_configuration
        meta.parse()
      File "/home/qba/.local/lib/python2.7/site-packages/setuptools/config.py", line 382, in parse
        section_parser_method(section_options)
      File "/home/qba/.local/lib/python2.7/site-packages/setuptools/config.py", line 355, in parse_section
        self[name] = value
      File "/home/qba/.local/lib/python2.7/site-packages/setuptools/config.py", line 173, in __setitem__
        value = parser(value)
      File "/home/qba/.local/lib/python2.7/site-packages/setuptools/config.py", line 430, in _parse_version
        version = self._parse_attr(value)
      File "/home/qba/.local/lib/python2.7/site-packages/setuptools/config.py", line 305, in _parse_attr
        module = import_module(module_name)
      File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
        __import__(name)
    ImportError: No module named vorta._version
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

**Comment by m3nu** [comment]
Python 2.7 won't work. 😬 Has been dead for years now. Don't use it.

**Comment by m3nu** [comment]
You may need `pip3`, if your system ships with both.

**Comment by hparlak** [comment]
I have this problem too on Fedora 36. Downgrading borg to 1.2.0 solved the problem. Waiting for a fix.

**Comment by m3nu** [comment]
> Waiting for a fix.

Fix is already in the master branch, @hparlak . Test like this:

> You can also test the current master branch using Pip, as described here: https://vorta.borgbase.com/contributing/#testing-the-current-development-version

I'll do a release once it's clear the issue is solved in the master branch. Thanks!

**Comment by ThomasWaldmann** [comment]
I also had the "hang at the end" issue with vorta 0.8.6 and (homebrew) borg 1.2.1.

It hung when / after processing the final file. Maybe this is due to my fix of the final json progress datastructure.

I tested the 0.8.7rc dmg on macOS M1 (note: it internally still says it is 0.8.6, but now includes borg 1.2.1, not 1.2.0 any more as the real 0.8.6 did). It now works as expected and tells "Backup finished..." instead of hanging.

**Comment by ThomasWaldmann** [comment]
BTW, guess you could use the progress msg with finished == True to clear the progress display.

Currently, it tells:
```
Backup finished with warnings. See logs for details.
/Users/tw/.../thelastfileitbackedup
```

**Comment by m3nu** [comment]
Thanks for testing. Yeah, we may use the new `finished` parameter in the future.

Will do a formal release tomorrow morning.

**Comment by m3nu** [comment]
Fixed in 0.8.7: https://github.com/borgbase/vorta/releases/tag/v0.8.7
